services/api-server/src/simcore_service_api_server/services_rpc/catalog.py

```diff
@@ -113,3 +114,33 @@ async def get(
             service_key=name,
             service_version=version,
         )
+
+    @_exception_mapper(
+        rpc_exception_map={
+            CatalogItemNotFoundError: ProgramOrSolverOrStudyNotFoundError,
+            CatalogForbiddenError: ServiceForbiddenAccessError,
+            ValidationError: InvalidInputError,
+        }
+    )
+    async def get_service_ports(
+        self,
+        *,
+        product_name: ProductName,
+        user_id: UserID,
+        name: ServiceKey,
+        version: ServiceVersion,
+    ) -> list[ServicePortGet]:
+        """Gets service ports (inputs and outputs) for a specific service version
+
+        Raises:
+            ProgramOrSolverOrStudyNotFoundError: service not found in catalog
+            ServiceForbiddenAccessError: no access rights to read this service
+            InvalidInputError: invalid input parameters
+        """
+        return await catalog_rpc.get_service_ports(
+            self._client,
+            product_name=product_name,
+            user_id=user_id,
+            service_key=name,
+            service_version=version,
+        )
```

Comment:
so I get the point of the CSR pattern. but if I remember well, directly calling functions from functions in python is expensive and does not help when following bugs... Also just a thought, but I would skip these kind of things. Especially in the API-server where you do not need that additional encapsulation at all since it is not (yet?) a monolith